src/campaign/components/ItemCard/TagGroup.tsx

```diff
@@ -0,0 +1,55 @@
+import {
+  Wrap,
+  WrapProps,
+  useBoolean,
+  IconButton,
+  BadgeProps,
+  Tag,
+} from "@chakra-ui/react";
+import { useEffect, useState } from "react";
+import { CgMoreAlt, CgMore } from "react-icons/cg";
+
+type TagGroupProps = {
+  displayLimit?: number;
+  tags: string[];
+} & Pick<BadgeProps, "variant" | "colorScheme"> &
+  WrapProps;
+
+const TagGroup: React.FC<TagGroupProps> = ({
+  displayLimit,
+  tags,
+  variant,
+  colorScheme,
+  ...wrapProps
+}) => {
+  const [showAll, setShowAll] = useBoolean(false);
```

Comment:
How's this different from useState(false)?

Reply:
It just gives some more useful functions specifically tailored to booleans. These are in the setShowAll obect.

Below I'm using `onClick={setShowAll.toggle}` instead of `onClick={() => setShowAll(!showAll)}` it also provides one-way functions like setShowAll.off and setShowAll.on